cmd/cli/plugin/feature/list.go

```diff
@@ -7,187 +7,206 @@ import (
 	"context"
 	"fmt"
 
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	crClient "sigs.k8s.io/controller-runtime/pkg/client"
-
 	"github.com/spf13/cobra"
 
+	corev1alpha2 "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
 	"github.com/vmware-tanzu/tanzu-framework/cli/runtime/component"
 	"github.com/vmware-tanzu/tanzu-framework/featuregates/client/pkg/featuregateclient"
 )
 
-var featuregate, outputFormat string
-var activated, deactivated, unavailable, extended bool
+var (
+	featuregate, outputFormat     string
+	activated, deactivated        bool
+	extended, includeExperimental bool
+)
 
-// FeatureListCmd is for activating Features
+// FeatureListCmd is for listing features in (a) featuregate(s).
 var FeatureListCmd = &cobra.Command{
 	Use:   "list",
-	Short: "List Features",
+	Short: "List features",
 	Args:  cobra.NoArgs,
 	Example: `
-	# List a clusters Features
+	# List feature(s) referenced in featuregate(s)
 	tanzu feature list --activated
-	tanzu feature list --unavailable
 	tanzu feature list --deactivated`,
-	RunE: featureList,
+	RunE: printFeatures,
 }
 
 func init() {
-	FeatureListCmd.Flags().BoolVarP(&extended, "extended", "e", false, "Include extended output. Higher latency as it requires more API calls.")
-	FeatureListCmd.Flags().StringVarP(&featuregate, "featuregate", "f", "tkg-system", "List Features gated by a particular FeatureGate")
-	FeatureListCmd.Flags().BoolVarP(&activated, "activated", "a", false, "List only activated Features")
-	FeatureListCmd.Flags().BoolVarP(&deactivated, "deactivated", "d", false, "List only deactivated Features")
-	FeatureListCmd.Flags().BoolVarP(&unavailable, "unavailable", "u", false, "List only Features specified in the gate but missing from cluster")
+	FeatureListCmd.Flags().BoolVarP(&extended, "extended", "e", false, "Include extended output")
+	FeatureListCmd.Flags().StringVarP(&featuregate, "featuregate", "f", "", "List features gated by specified FeatureGate")
+	FeatureListCmd.Flags().BoolVarP(&activated, "activated", "a", false, "List only activated features")
+	FeatureListCmd.Flags().BoolVarP(&deactivated, "deactivated", "d", false, "List only deactivated features")
 	FeatureListCmd.PersistentFlags().StringVarP(&outputFormat, "output", "o", "", "Output format (yaml|json|table)")
+	FeatureListCmd.Flags().BoolVarP(&includeExperimental, "include-experimental", "x", false, "Include experimental features in list")
 }
 
-// FeatureInfo is a struct that holds Feature information
+// FeatureInfo is a struct that holds Feature information an if it can be listed by plugin.
 type FeatureInfo struct {
-	Name        string
-	Maturity    string
-	Description string
-	Activated   bool
-	Available   bool
-	Immutable   bool
+	Name         string
+	Description  string
+	Stability    corev1alpha2.StabilityLevel
+	Immutable    bool
+	Discoverable bool
+	FeatureGate  string
+	Activated    bool
+	Listable     bool
```

Comment:
what is Listable for?

Reply:
I should probably rename it since it's confusing. I'm thinking `showInList` or something.

Reply:
I've changed it to `ShowInList` (exported like all other fields). I'll resolve this for now and you can suggest something else in another review comment.